### ISSUE TITLE
URGENT - Tim and Rich Issue with NCSM Farm Journal

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#676]
In GitLab by @kdjstudios on Feb 8, 2018, 14:41

**Submitted by:** Tim

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2018-02-08-41459/conversation

**Server:** Internal

**Client/Site:** NCSM

**Account:** Farm Journal

**Issue:** 

Rich says that under NCSM Farm Journal has a single invoice in either Dec or Jan, but the AR snapshot had a different balance.  He needs to be able to explain the difference to the auditors, so we need for Aavaz to check it out ASAP.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 8, 2018, 14:41

@smasih 

I believe this may be also linked to another ticket, if I find it I will link it here.

Possible link?: #609

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 8, 2018, 14:53

My email to Richard:

> Hello Richard,
> 
>  
> 
> Tim communicated with me today that  under NCSM Farm Journal has a single invoice in either Dec or Jan, but the AR snapshot had a different balance. And that you are in need to be able to explain the difference to the auditors.
> 
>  
> 
> Could you please elaborate on which Farm Journal account, invoice, billing cycle, etc that is having the issue? I have already informed the support team, but these details will help narrow down the research and time the support team has to perform to find and resolve the issue.  Thank you.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 8, 2018, 14:54

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 8, 2018, 15:08

Tim was able to clarify, this is for the Farm Journal Account in Webster City. 

124-N0017 NCSM - Farm Journal Media

If I am looking at their billing history page correctly, they billed three invoices in October, but none in November. Also it looks as though the billing cycle date, changed from the 1st of the month, to the 30th of the month. Then also, now the billing dates seem sporadic. 

![Untitled](/uploads/9c078befb714e06f4c972d4e2400e94d/Untitled.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 8, 2018, 16:59

Received email reply from Richard below:

> The issue is that the invoice for Farm Journal # 001-25610 attached, is a total of $ 277,506.14.  From looking at the invoice, it was generated on 12/27/17.  When I compare this invoice amount with the AR snapshot report at 12/31/17 which shows a balance of $ 193,977.34, there is a difference of $ 83,528.80.
> 
> Our outside accountants have selected the Farm Journal to confirm as part of their audit, and I need to understand this difference. 
> 
> There was a payment of $ 66,609.99  received on 12/27/12, however, this payment was  not entered until 1/8/18 and would not have been reflected in either the invoice 001-25610 or the AR snapshot at 12/31/17.
> 
> What should the correct balance as entered through 12/31/17 be, the $ 277,506.14 or the $ 193,977.34?
> 
> Please expedite this request.

[Farm_Journal_invoice_001-25610_01-04-18.pdf](/uploads/e147231684e4de7c10c389c25a07b230/Farm_Journal_invoice_001-25610_01-04-18.pdf)

[NCSM_ar_snapshot__12.31.2017.xlsx](/uploads/9e9d15a427920a5c0475e86b45788118/NCSM_ar_snapshot__12.31.2017.xlsx)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 8, 2018, 17:06

Judging from differences from Tim's input and what richard is now emailing. The Farm Journal account being referenced is not the one on Webstery City, but a different one. It appears to be possibly the following account according to the invoice:

Farm Journal Media's Webster City IA call center
NCSM 001-0017 Farm journal Media

It also appears that the invoice he attached is not from 12/27/17; but instead from 1/4/17.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 8, 2018, 17:12

Aavaz team please look into why there is a difference between the AR snapshot report and the invoice at that point in time.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Feb 9, 2018, 24:03

@kdjstudios 

The last transaction was an Invoice ([+001-25493+]) of [-$84786.03-] on `12/27/2017` which left the account with balance of **$277506.14**

Payment of `$66,609.99` never came in before `01/09/2018 20:29` so it's upto the accounting standards whether we want to consider it or not as it's paid date is `12/27/2017`

Please look for the attached [account ledger report](/uploads/e5f74cb025606798302efcdb5bbec609/Farm_journal_Media_-_Ledger_-_09February2018__1_.csv) sorted on invoice and payment **generation/creation date.**

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Feb 9, 2018, 01:33

AR Snapshot is having $193977.34 till 12/31/17 while the balance should be 277506.14 which is what reflecting on the History and invoice.

AR Snapshot runs at 12AM for first of every month and captures the detail, so we had a window of 5 days in which there could have been any other transactions on the account. We tried finding deleted payment in application logs but didn't find any within that time frame.   


In short, we are not sure why there is a difference in AR snapshot and Invoice balance.

**Comment by kdjstudios** [comment]
In GitLab by @abakshi on Feb 9, 2018, 06:07

@pchaudhary 

As discussed with @tim.traylor in the huddle:

This is critical for accounting audits and we have to explain to the auditors to retain their trust.

Key points to check:
* Bill payment - Checked. Will recheck.
* Charges / Late fees/ Stage fees getting applied - Will check
* Use spreadsheet to go through the balances
* Is AR Snapshot matching the account details?

Open Questions to answer:
* Why there is a difference? 
* What's the correct balance?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 9, 2018, 07:54

@abakshi @pchaudhary @tim.traylor 

So do we have any response for Richard today? It appears that this is marked for approval, but has no answer or solution yet?

Removing approval, marking for development.

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Feb 13, 2018, 09:09

Please use the backup from the end of January to get the invoice balances as of month end.  Then take those balances and compare them to the current balance on the AR snapshot from the end of January.  The two should match exactly, and if there are any differences maybe they can help us in identifying why the two things are off.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Feb 14, 2018, 07:42

@kdjstudios 

We did what Tim asked us to do and found there was no difference in account balance and the ar snapshot reports.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 14, 2018, 10:33

Update sent to Richard:

> Thank you for an update. The support team is still looking into this. We do want to clarify the following.
> 
> The AR Snapshot runs at 12AM for first of every month and captures the detail. This would mean any invoices, payments, fees, etc entered after then point (even though it may be dated prior to this) will not be apart of the AR Snapshot report. 
> 
> That is what appears to have happened as the last transaction was an Invoice (001-25493) of $84786.03 on 12/27/2017 which left the account with balance of $277506.14
> 
> Payment of $66,609.99 never came in before 01/09/2018 20:29 but it's paid date is 12/27/2017 - (Which is why it was not on the Snapshot). Please look at the attached account ledger report sorted on invoice and payment generation/creation date.
> 
> Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Feb 15, 2018, 05:35

> The AR Snapshot runs at 12AM for first of every month and captures the detail. This would mean any invoices, payments, fees, etc entered after then point (even though it may be dated prior to this) will not be apart of the AR Snapshot report.

that's right.



> That is what appears to have happened as the last transaction was an Invoice (001-25493) of $84786.03 on 12/27/2017 which left the account with balance of $277506.14

As NCSM is a manual site, its good to assume that each invoice was on draft stage until next invoice came in and draft invoice does not affect account balance. Just as `12/27/2017` invoice was not finalized until 2^nd JAN

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 15, 2018, 08:02

@smasih 

Just to confirm in summary, this was caused by invoices and payments not being finalized until after the AR snapshot was taken correct?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Feb 15, 2018, 23:46

> Just to confirm in summary, this was caused by invoices and payments not being finalized until after the AR snapshot was taken correct?

We cannot confirm that but yaa, that's our best guess.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Feb 26, 2018, 24:23

@kdjstudios 

we shared what we had on this issue, is there anything else we need to do..?

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Feb 26, 2018, 05:48

@kdjstudios @smasih Hi Guys - Aside from keeping the additional backups, are there other changes we should plan so that we can ALWAYS explain what happened?  Tim

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Feb 26, 2018, 06:19

*  Add sub-collection to invoice that tracks the datetime, amount, and payment to show what amount of what payment(s) got applied to the invoice.
*  Do not allow payments to be deleted! They should be reversed so we keep the history.
*  @smasih Sumeet is going to detail another scenario where a draft invoice may be an issue and we need to track additional info.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Feb 27, 2018, 06:43

@kdjstudios  

> @smasih Sumeet is going to detail another scenario where a draft invoice may be an issue and we need to track additional info.

I thought over it and @tim.traylor was right on the first place. If we are not allowing drafts to get associated with closed billing cycles we won't get trapped in missing draft invoice issue.

As per Tim's suggestion we did the required changes on merge http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/719#note_16008

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Mar 6, 2018, 06:55

We can delete additional fees on account level #729. Deactivate button to be replaced with delete button of the additional fees on account.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 25, 2018, 14:16

@smasih @pchaudhary @amishra @tim.traylor 

I just want to confirm that this has been completed in different tickets correct?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 26, 2018, 08:35

@kdjstudios 

The core issue was that manual invoices were being associated with the closed billing cycles.

and yes, kyle changes are live and working.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 26, 2018, 08:38

@kdjstudios 

no sorry, I read it again and the core issue was resolved but we still need to work on the preventive measures as described by @tim.traylor  in [this comment](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/676#note_15936)  

marking the ticket for development

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 26, 2018, 08:38

assigned to @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Sep 26, 2018, 11:05

@smasih 

Thank you, May we get an estimate on what is still needed. Then lets discuss this with @tim.traylor to get an updated priority level.

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Sep 27, 2018, 01:57

assigned to @smasih and unassigned @pchaudhary

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 28, 2018, 04:23

@kdjstudios 


> Add sub-collection to invoice that tracks the datetime, amount, and payment to show what amount of what payment(s) got applied to the invoice

This will be completed under #1090 Detail Open AR Report 
 * We will be maintaining a collection of payment amount used to pay off a particular invoice

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Sep 28, 2018, 04:23

marking it as a secondary

**Comment by kdjstudios** [comment]
In GitLab by @tim.traylor on Oct 15, 2018, 06:10

@smasih Hi Sumeet, I don't think the new payment tracking collection was part of the specs for the daily AR.  In your comment above, did you just mean they could be done at the same time?

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Oct 16, 2018, 08:56

@tim.traylor 


payment tracking collection is already released and working (but that is only for manual payments because they were the only one which was getting deleted.)


Then there will be another collection "Payment Invoice Mapping" which will contain amount used against per invoice. **We need this for our daily AR report so we will be releasing them together.** http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1090#note_27761

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 14, 2018, 15:03

@smasih @amishra @pchaudhary @tim.traylor 

May we close this ticket? It would appear this will be released with the #1090 ticket correct?

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 21, 2018, 06:57

@kdjstudios 

Yes, you can close the ticket.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 21, 2018, 08:33

marking for close. Will close this once #1090 is closed.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 27, 2018, 09:38

closed
